src/client.rs

```diff
@@ -14,27 +14,30 @@ pub struct WebPushClient {
 impl WebPushClient {
     pub fn new() -> WebPushClient {
         let mut builder = Client::builder();
-        builder.keep_alive(true);
+        builder.pool_max_idle_per_host(std::usize::MAX);
```

Comment:
Are you sure about this? This is a lot of connections...

Reply:
This was because `keep_alive` [was deprecated](https://github.com/hyperium/hyper/blob/dfa1bb291d71978820e8077be94c1d5837935bbc/src/client/client.rs#L878), and the deprecation message was suggesting using `pool_max_idle_per_host` (which it does under the hood). `usize::MAX` [is the default value and means "no limit"](https://docs.rs/hyper/0.14.4/hyper/client/struct.Builder.html#method.pool_max_idle_per_host). 
If you look at the code for the [`keep_alive`] function, setting it to `true` does exactly that. But if you want to introduce a hard limit, now is the time!

In any case since this is `hyper`-specific, it is removed in your other pull request.

Reply:
Ah, no, this is interesting. I'm just surprised you'd have a limit of `18446744073709551615` connections. You'd first just hit your system `ulimit`, which is about 1024 or something similar, depending on your setup. I don't think you need to set this anymore, but please check from Hyper what is the correct way of having keep-alive set for http1.

Btw, hyper will be gone if we merge my PR about being runtime-independent, and I'd like to cut a new major version when we release the new ece support and runtime-independence, so we have some time to test.

Reply:
That's what I was saying. Probably not worth the time since it's going to be removed. By the way I have the version that works on your fork (a lot of little conflicts on formatting) so since I already resolved them I can push it on another branch or even udpate this PR once the other one is merged somewhere.